### DISCUSSION TITLE
Win10 Toast Notification with Schedule buttton to pick a time

### DISCUSSION THREAD

**Discussion by VarunChitra** [Windos/BurntToast#179] [Q&A]
**_Function of Script_**:  1) It displays Notification with 2 buttions (Snooze & Install Now). 
 -> Snooze button - is for snoozing notification for specified time.
 -> Install Now Button - will trigger a Powershell Script to run in background to install a patch.

**_Question_**: How do we add schedule button to pick a time & date(calender) and pass it to Script to run at scheduled time using Task Scheduler.

``
Install-Module -Name BurntToast
Install-module -Name RunAsUser

#Checking if ToastReboot:// protocol handler is present
New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT -erroraction silentlycontinue | out-null

$ProtocolHandler = get-item 'HKCR:\ToastReboot' -erroraction 'silentlycontinue'
if (!$ProtocolHandler) {
    #create handler for reboot
    New-item 'HKCR:\ToastReboot' -force
    set-itemproperty 'HKCR:\ToastReboot' -name '(DEFAULT)' -value 'url:ToastReboot' -force
    set-itemproperty 'HKCR:\ToastReboot' -name 'URL Protocol' -value '' -force
    new-itemproperty -path 'HKCR:\ToastReboot' -propertytype dword -name 'EditFlags' -value 2162688
    New-item 'HKCR:\ToastReboot\Shell\Open\command' -force
    #set-itemproperty 'HKCR:\ToastReboot\Shell\Open\command' -name '(DEFAULT)' -value 'C:\Windows\System32\shutdown.exe -r -t 00' -force
    set-itemproperty 'HKCR:\ToastReboot\Shell\Open\command' -name '(DEFAULT)' -value 'c:\windows\system32\WindowsPowerShell\v1.0\powershell.exe C:\Patching_Automation\Install_AvailableUpdate.ps1' -force

}
 
    #$heroimage = New-BTImage -Source 'https://media.giphy.com/media/eiwIMNkeJ2cu5MI2XC/giphy.gif' -HeroImage
    $heroimage = New-BTImage -Source 'C:\Patching_Automation\2.png' -HeroImage
    $Text1 = New-BTText -Content  "Message from IT"
    $Text2 = New-BTText -Content " Windows Feature Update will be Installed on your computer at $(get-date). Please select if you'd like to Install now, or snooze this message."
    $Button = New-BTButton -Content "Snooze" -snooze -id 'SnoozeTime'
    $Button2 = New-BTButton -Content "Install Now" -Arguments "ToastReboot:" -ActivationType Protocol
    $5Min = New-BTSelectionBoxItem -Id 1 -Content '1 minutes'
    $10Min = New-BTSelectionBoxItem -Id 10 -Content '10 minutes'
    $1Hour = New-BTSelectionBoxItem -Id 60 -Content '1 hour'
    $4Hour = New-BTSelectionBoxItem -Id 240 -Content '4 hours'
    $1Day = New-BTSelectionBoxItem -Id 1440 -Content '1 day'
    $Items = $5Min, $10Min, $1Hour, $4Hour, $1Day
    $SelectionBox = New-BTInput -Id 'SnoozeTime' -DefaultSelectionBoxItemId 10 -Items $Items
    $action = New-BTAction -Buttons $Button, $Button2 -inputs $SelectionBox
    $Binding = New-BTBinding -Children $text1, $text2 -HeroImage $heroimage
    $Visual = New-BTVisual -BindingGeneric $Binding
    $Content = New-BTContent -Visual $Visual -Actions $action
    Submit-BTNotification -Content $Content
``

**Comment by unknownactor487762** [comment]
Not task scheduler, but you can get a visual time picker (easy to add date too)
Like this

Add-Type -AssemblyName System.Windows.Forms
$form = New-Object System.Windows.Forms.Form
$form.Text = "Select Reboot Time"
$form.Size = New-Object System.Drawing.Size(300,150)
$form.StartPosition = "CenterScreen"

#DateTimePicker control with UpDown style for hour and minute
$dateTimePicker = New-Object System.Windows.Forms.DateTimePicker
$dateTimePicker.Location = New-Object System.Drawing.Point(50,20)
$dateTimePicker.Size = New-Object System.Drawing.Size(200,50)
$dateTimePicker.Format = [System.Windows.Forms.DateTimePickerFormat]::Custom
$dateTimePicker.CustomFormat = "hh:mm tt"  # "tt" adds AM/PM indicator
$dateTimePicker.ShowUpDown = $true

# Create a button to confirm selected time
$button = New-Object System.Windows.Forms.Button
$button.Location = New-Object System.Drawing.Point(100,80)
$button.Size = New-Object System.Drawing.Size(100,30)
$button.Text = "OK"
$button.DialogResult = [System.Windows.Forms.DialogResult]::OK
$form.AcceptButton = $button

# Add controls to the form
$form.Controls.Add($dateTimePicker)
$form.Controls.Add($button)

# Display the form and wait for user input
$result = $form.ShowDialog()

# Process the user's choice
if ($result -eq [System.Windows.Forms.DialogResult]::OK) {
    $selectedTime = $dateTimePicker.Value
    $currentTime = Get-Date
    $timeDifference = ($selectedTime - $currentTime).TotalSeconds
    
    # Calculate the shutdown time in seconds from now
    $shutdownInSeconds = [math]::Round($timeDifference, 0)
    
    # Schedule system reboot at the specified time using shutdown command
    $shutdownCommand = "shutdown /r /f /t $shutdownInSeconds /c 'Scheduled reboot at $selectedTime'"
    Invoke-Expression -Command $shutdownCommand
}

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
